specification/app/resource-manager/Microsoft.App/preview/2024-02-02-preview/ContainerApps.json

```diff
@@ -1409,6 +1419,34 @@
           "type": "string"
         }
       }
+    },
+    "IdentitySettings": {
+      "description": "Optional settings for a Managed Identity that is assigned to the Container App.",
+      "type": "object",
+      "required": [
+        "identity"
+      ],
+      "properties": {
+        "identity": {
+          "description": "The resource ID of a user-assigned managed identity that is assigned to the Container App, or 'system' for system-assigned identity.",
+          "type": "string"
+        },
+        "lifecycle": {
+          "description": "The lifecycle stages of a Container App at which the Managed Identity is available.",
+          "enum": [
```

Comment:
Can you please explain how the api caller will be using the "IdentitySettings" property ? Is the callers responsibility to ensure that the identity is present at these stages ?  What happens if it is not present ?

Reply:
This is where the caller gets to choose the stages at which they want the identity to be available. Users have requested the ability to control when managed identity is available during a Container App lifecycle to improve least-privilege. For example, a user can use this to have an identity available only during the init stage to do some initialization work, and not have it available during the main stage. I updated the description to make it clearer that this setting gives users this choice.

Reply:
@raosuhas please let me know if that makes sense.